tools/PI/DevHome.PI/Helpers/WindowHooker`1.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using System.Threading;
+using DevHome.PI.Helpers;
+using DevHome.PI.Models;
+using Serilog;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using Windows.Win32.UI.WindowsAndMessaging;
+
+namespace DevHome.PI.Helpers;
+
+internal abstract class WindowHooker<T>

```

Comment:
this needs a comment regarding it's intended use case and purpose.

Reply:
I'll leave this for @aeloros to do on his next check-in.